composer/trainer/trainer.py

```diff
@@ -1836,6 +1842,59 @@ def predict(self, dataloader: Union[DataLoader, DataSpec], subset_num_batches: i
             subset_num_batches (int, optional): If specified, only perform model prediction
                 on this many batches. This parameter has no effect if it is greater than ``len(dataloader)``.
                 If ``-1``, then the entire loader will be iterated over. (default: ``-1``)
+            return_outputs (bool, optional): If True (the default), then prediction outputs will be (recursively)
+                moved to cpu and accumulated into a list. Otherwise, prediction outputs are discarded after each
+                batch.
+
+                .. note::
+
+                    If collecting all outputs from the dataloader would exceed available memory, we recommend
+                    to set `return_outputs` to False and use your own callback to write the outputs from each
+                    prediction batch to a file. For example:
+
+                    .. testsetup::
+
+                        predict_dl = train_dataloader
+
+                    .. testcode::
```

Comment:
should we just make this its won callback and backlink it here?

Reply:
if not, we should move this into the main body of the docstrings as a "There are two ways to access the outputs of `predict`. With `return_outputs=True`, you can use the list of answers and save them to disk yourself, e.g. `outputs = trainer.predict(..); torch.save(outputs, ..)`.

Or if the total outputs are too large, use `return_outputs=False`, and add the predictionCallback to save things to disk or object store  with each batch.

Reply:
I'd rather leave it as a docstring for now until we can write a robust callback. But agreed, I'll put this high level in the docstring.

Reply:
Updated the docstring.